wandb/old/settings.py

```diff
@@ -49,9 +51,16 @@ def write_setting(settings, settings_path, persist):
             if not settings.has_section(section):
                 Settings._safe_add_section(settings, Settings.DEFAULT_SECTION)
             settings.set(section, key, str(value))
+
             if persist:
-                with open(settings_path, "w+") as f:
-                    settings.write(f)
+                # write a temp file and then move it to the settings path
+                with tempfile.NamedTemporaryFile(
```

Comment:
I think the tempfile should be created on the same filesystem as the target and then use os.rename.